Src/VimWpf/Implementation/BlockCaret/BlockCaret.cs

```diff
@@ -338,36 +376,45 @@ private FormattedText CreateFormattedText()
         private Size CalculateCaretSize(out string caretCharacter)
         {
             caretCharacter = "";
-            var defaultWidth = FormattedText.Width;
 
+            var defaultWidth = _formattedText.Width;
             var width = defaultWidth;
             var height = _textView.LineHeight;
 
             if (IsRealCaretVisible)
             {
-                // Get the size of the character to which we need to paint
-                // the caret.  Special case tab here because it's too big.
-                // When there is a tab we use the default height and width.
+                // Get the caret string and caret width.
                 var line = TextViewLineContainingCaret;
                 height = line.Height;
                 var point = _textView.Caret.Position.BufferPosition;
                 if (point.Position < _textView.TextSnapshot.Length)
                 {
                     var pointCharacter = point.GetChar();
-                    if (pointCharacter != '\t'
-                        && !_controlCharUtil.IsDisplayControlChar(pointCharacter))
+                    if (_controlCharUtil.TryGetDisplayText(pointCharacter, out string caretString))
+                    {
+                        // Handle control character notation.
+                        caretCharacter = caretString;
+                        width = line.GetCharacterBounds(point).Width;
+                    }
+                    else if (Char.IsHighSurrogate(pointCharacter) &&

```

Comment:
Think we should add a helper to `SnapshotPoinUtil` for this. Pretty sure we've written this code in a few places.

Reply:
Converted it to use `SnapshotCodePoint.CodePointInfo`.